api/V1/Recording.ts

```diff
@@ -471,6 +475,85 @@ export default (app: Application, baseUrl: string) => {
     })
   );
 
+  /**
+   * @api {get} /api/v1/recordings/:id/thumbnail Gets a thumbnail png for this recording
+   * @apiName RecordingThumbnail
+   * @apiGroup Recordings
+   * @apiDescription Gets a thumbnail png for this recording in Viridis palette
+   *
+   * @apiSuccess {file} file Raw data stream of the png.
+   * @apiUse V1ResponseError
+   */
+  app.get(
+    `${apiUrl}/:id/thumbnail`,
+    [param("id").isInt()],
+    middleware.requestWrapper(async (request, response) => {
+      const rec = await models.Recording.findByPk(request.params.id, {
+        attributes: ["rawFileKey", "id"]
+      });
+      if (!rec) {
+        return responseUtil.send(response, {
+          statusCode: 400,
+          messages: ["Failed to get recording."]
+        });
+      }
+      const mimeType = "image/png";
+      const filename = `${rec.id}-thumb.png`;
+
+      if (!rec.rawFileKey) {
+        throw new ClientError("Rec has no raw file key.");
+      }
+
+      const s3 = modelsUtil.openS3();
+      const params = {
+        Bucket: config.s3.bucket,
+        Key: `${rec.rawFileKey}-thumb`
+      };
+
+      s3.getObject(params, function (err, data) {
+        if (err) {
+          log.error("Error getting thumbnail from s3 %s", rec.id);
+          log.error(err.stack);
+          return responseUtil.send(response, {
+            statusCode: 400,
+            messages: ["No thumbnail exists"]
+          });
+        }
+
+        if (!request.headers.range) {
+          response.setHeader(
+            "Content-disposition",
+            "attachment; filename=" + filename
+          );
+          response.setHeader("Content-type", mimeType);
+          response.setHeader("Content-Length", data.ContentLength);
+          response.write(data.Body, "binary");
+          return response.end(null, "binary");
+        }
+
+        const range = request.headers.range;
```

Comment:
Why would we ever have a range request for a thumbnail image?  We're always going to just server the whole thing, so should never get to here

Reply:
true, i thought it was at the browsers disposition to use range or not. But by not setting accept-ranges we are saying we dont support it

Reply:
Yeah, we shouldn't support it for that endpoint, it only makes sense in terms of streaming media formats, of which png is not one.